src/lib.rs

```diff
@@ -448,6 +452,24 @@ impl ExtendedFullViewingKey {
             dk: self.dk.derive_child(i_l),
         })
     }
+
+    pub fn address(
+        &self,
+        j: DiversifierIndex,
+    ) -> Result<(DiversifierIndex, PaymentAddress<Bls12>), ()> {
+        let (j, d_j) = match self.dk.diversifier(j) {
+            Ok(ret) => ret,
+            Err(()) => return Err(()),
+        };
+        match self.fvk.vk.into_payment_address(d_j, &JUBJUB) {
+            Some(addr) => Ok((j, addr)),
+            None => Err(()),
+        }
+    }
+
+    pub fn default_address(&self) -> Result<(DiversifierIndex, PaymentAddress<Bls12>), ()> {
+        self.address(DiversifierIndex::new())
```

Comment:
Is this assuming the 0 diversifier index will return a valid diversifier?

Reply:
`address()` calls `dk.diversifier()`, which finds and returns the first valid difersifier index starting from the given index. So `default_address()` will use the smallest diversifier index that results in a valid diversifier.

Reply:
right..